src/main/java/webserver/http/request/HttpRequestUrl.java

```diff
@@ -0,0 +1,71 @@
+package webserver.http.request;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import webserver.http.exception.InvalidHttpRequestException;
+
+public class HttpRequestUrl {
+    private static final String PATH_AND_PARAMETER_DIVIDER = "?";
+    private static final String PARAMETERS_DIVIDER = "&";
+    private static final String PARAMETER_DIVIDER = "=";
+    private static final String EMPTY_CONTENT = "";
+
+    private final String requestUrl;
+    private final Map<String, String> parameters = new LinkedHashMap<>();
+
+    public HttpRequestUrl(String requestUrl) {
+        this.requestUrl = requestUrl;
+        String parameterLine = requestUrl.substring(requestUrl.indexOf(PATH_AND_PARAMETER_DIVIDER) + 1);
+        parseParameter(requestUrl, parameterLine);
+    }
+
+    private void parseParameter(String requestUrl, String parameterLine) {
+        if (requestUrl.contains(PATH_AND_PARAMETER_DIVIDER)) {
+            String[] splitedParameters = parameterLine.split(PARAMETERS_DIVIDER);
+
+            for (String splitedParameter : splitedParameters) {
+                String[] keyAndValue = splitedParameter.split(PARAMETER_DIVIDER);
+                parameters.put(keyAndValue[0], keyAndValue[1]);
+            }
+            validate(parameters, splitedParameters.length);
+        }
+    }
+
+    private void validate(Map<String, String> parameters, int originalLength) {
+        if (parameters.size() != originalLength) {
+            throw new InvalidHttpRequestException("겹치는 key가 있습니다.");
+        }
+
+        if (parameters.containsKey(EMPTY_CONTENT)) {
+            throw new InvalidHttpRequestException("정보가 없는 key가 있습니다.");
+        }
+    }
+
+    public String getPath() {
+        if (hasParameters()) {
+            return requestUrl.substring(0, requestUrl.indexOf(PATH_AND_PARAMETER_DIVIDER));
+        }
+        return requestUrl;
+    }
+
+    public boolean isSamePath(String path) {
+        return getPath().equals(path);
+    }
+
+    private boolean hasParameters() {
+        return requestUrl.contains(PATH_AND_PARAMETER_DIVIDER);
+    }
+
+    public boolean isStaticFile() {
+        return StaticFiles.endsWith(requestUrl);
+    }
+
+    public Map<String, String> getParameters() {
+        return parameters;
+    }
```

Comment:
query string parameter에 대해서는 불변성을 보장해주는게 좋을 것 같아요.
실제 요청이 아닌데 변질 될 수 있는 위험이 있네요.

Reply:
현재 getParameters는 Test에만 사용되는 메서드입니다. 
다만 무분별하게 public으로 열어줘서 실제 요청을 변질시킬 수 있게 되어 있습니다.

실제 테스트에서 get을 두번 활용하여 expected와 값을 비교하기에 아래와 같이 메서드를 만들어 노출을 최소화 했습니다.
```suggestion
    protected Set<String> getKeySet() {
        return Collections.unmodifiableSet(parameters.keySet());
    }

    protected String getValueBy(String key) {
        return parameters.get(key);
    }
```

기존 getParameters()는 제거하였습니다.

미쳐 지나칠 수 있던 부분에 대한 리뷰 감사합니다😬👍